mesonbuild/environment.py

```diff
@@ -36,6 +36,7 @@
     is_object,
     is_source,
 )
+from .compilers.cuda import CudaCompiler
```

Comment:
I don't like the idea of pulling CudaCompiler knowledge into environment like this. As seen in the CI, it breaks test_setup_loaded_modules, and in general I think this is a layering violation. I'd actually expect this to be converted somewhere in the backend, maybe.

Reply:
Yes, importing CudaCompiler into `environment` is not a good idea. But CudaCompiler has no idea of the handling of `$LDFLAGS` in `environment`, so where should the handling be put?

Reply:
I have moved the logic to `detect.py` where `CudaCompiler` is already imported, so there is no extra module loaded now.